docs/specs/xref.yml

```
@@ -554,104 +554,111 @@ outputs:
-      "uid": "b"
-    }
+  .errors.log: |
+    ["error","circular-reference","Build has identified file(s) referencing each other: 'docs/a.yml' --> 'docs/b.yml' --> 'docs/c.yml' --> 'docs/a.yml'","docs/a.yml"]
+    ["error","circular-reference","Build has identified file(s) referencing each other: 'docs/b.yml' --> 'docs/c.yml' --> 'docs/a.yml' --> 'docs/b.yml'","docs/b.yml"]
+    ["error","circular-reference","Build has identified file(s) referencing each other: 'docs/c.yml' --> 'docs/a.yml' --> 'docs/b.yml' --> 'docs/c.yml'","docs/c.yml"]
 ---
-# SDP cyclic xref reference whose display properties referring to another uid
-# a.json -> b.json -> a.json && b.json -> a.json -> b.json
+# circular reference within SDP UID definition a -> b -> a -> c
 inputs:
-  docfx.yml: |
-    template: {APP_BASE_PATH}data/template
-  docs/a.json: |
-    {
-      "$schema": "https://raw.githubusercontent.com/dotnet/docfx/v3/schemas/TestData.json",
-      "uid": "a",
-      "name": "name for a",
-      "inlineDescription": "description for a",
-      "summary": "Link to @b?displayProperty=summary",
-    }
-  docs/b.json: |
-    {
-      "$schema": "https://raw.githubusercontent.com/dotnet/docfx/v3/schemas/TestData.json",
-      "uid": "b",
-      "name": "name for b",
-      "inlineDescription": "description for b",
-      "summary": "Link to @a?displayProperty=summary",
-    }
-  docs/c.json: |
+  docfx.yml:
+  docs/a.yml: |
+    #YamlMime:TestData
+    uid: a
+    xref: b
+    anotherXref: c
+  docs/b.yml: |
+    #YamlMime:TestData
+    uid: b
+    xref: a
+  docs/c.yml: |
+    #YamlMime:TestData
+    uid: c
+  _themes/ContentTemplate/schemas/TestData.schema.json: |
     {
-      "$schema": "https://raw.githubusercontent.com/dotnet/docfx/v3/schemas/TestData.json",
-      "uid": "c",
-      "name": "name for c",
-      "inlineDescription": "description for c",
-      "summary": "Link to @a",
+      "properties": {
+          "uid": {
+            "type": "string",
+            "contentType": "uid"
+          },
+          "xref": {
+              "type": "string",
+              "contentType": "Xref"
+          },
+          "anotherXref": {
+              "type": "string",
+              "contentType": "Xref"
+          }
+      },
+      "xrefProperties": [
+          "xref"
+      ]
     }
 outputs:
-  docs/c.json: |
-    {"name":"name for c","summary":"<p>Link to <a href=\"a.json\">name for a</a></p>\n","inlineDescription":"description for c","uid":"c"}
   .errors.log: |
-    ["error","circular-reference","Build has identified file(s) referencing each other: 'docs/a.json' --> 'docs/b.json' --> 'docs/a.json'","docs/a.json"]
-    ["error","circular-reference","Build has identified file(s) referencing each other: 'docs/b.json' --> 'docs/a.json' --> 'docs/b.json'","docs/b.json"]
+    ["error","circular-reference","Build has identified file(s) referencing each other: 'docs/a.yml' --> 'docs/b.yml' --> 'docs/a.yml'","docs/a.yml"]
```

consider use `$"{uid} ({file}"` because there might be multiple uids in the same file
```suggestion
    ["error","circular-reference","Build has identified file(s) referencing each other: 'a (docs/a.yml)' --> 'docs/b.yml' --> 'docs/a.yml'","docs/a.yml"]
```

---

sounds reasonable